zeroconf.py

```diff
@@ -1608,20 +1610,19 @@ def find(cls, zc=None, timeout=5, interfaces=InterfaceChoice.All):
         return tuple(sorted(listener.found_services))
 
 
-def get_all_addresses(address_family):
-    return list(set(
-        addr['addr']
-        for iface in netifaces.interfaces()
-        for addr in netifaces.ifaddresses(iface).get(address_family, [])
-        if addr.get('netmask') != HOST_ONLY_NETWORK_MASK
-    ))
+def get_all_addresses():
+    addresses = []
+    for iface in ifaddr.get_adapters():
+        for addr in iface.ips:
+            if addr.network_prefix == HOST_ONLY_NETWORK_MASK & len(addr.ip.split(".")) == 4: addresses.append(addr.ip)
+    return addresses
```

Comment:
Would you mind restoring both the `list(set(...))` call (in order to avoid duplicates, not sure if the issue this was supposed to avoid can occur with `ifaddr`, better safe than sorry) and replacing the imperative `addresses = []; for ...: addresses.append()` form with generator expression as it was done before? I like it that way because I believe it's easier to read.